plugins/utils/ccc.nix

```
@@ -5,59 +5,252 @@
   pkgs,
   ...
 }:
+with lib;
 helpers.neovim-plugin.mkNeovimPlugin config {
   name = "ccc";
   originalName = "ccc.nvim";
   defaultPackage = pkgs.vimPlugins.ccc-nvim;
 
   maintainers = [ helpers.maintainers.JanKremer ];
 
-  settingsOptions = {
-    default_color = helpers.defaultNullOpts.mkStr "#000000" ''
-      The default color used when a color cannot be picked. It must be HEX format.
-    '';
+  settingsOptions =
+    let
+      listOfRawLua = with helpers.nixvimTypes; listOf strLua;
+      mapToRawLua = map helpers.mkRaw;
+    in
+    {
+      default_color = helpers.defaultNullOpts.mkStr "#000000" ''
+        The default color used when a color cannot be picked. It must be HEX format.
+      '';
 
-    highlight_mode =
-      helpers.defaultNullOpts.mkEnum
-        [
-          "fg"
-          "bg"
-          "foreground"
-          "background"
-        ]
-        "bg"
-        ''
-          Option to highlight text foreground or background. It is used to
-          `output_line` and `highlighter`.
-        '';
-
-    lsp = helpers.defaultNullOpts.mkBool true ''
-      Whether to enable LSP support. The color information is updated in the
-      background and the result is used by `:CccPick` and highlighter.
-    '';
+      inputs = mkOption {
+        type = listOfRawLua;
+        apply = mapToRawLua;
+        default = [ ];
+        example = [
+          "ccc.input.rgb"
+          "ccc.input.hsl"
+          "ccc.input.hwb"
+          "ccc.input.lab"
+          "ccc.input.lch"
+          "ccc.input.oklab"
+          "ccc.input.oklch"
+          "ccc.input.cmyk"
+          "ccc.input.hsluv"
+          "ccc.input.okhsl"
+          "ccc.input.hsv"
+          "ccc.input.okhsv"
+          "ccc.input.xyz"
+        ];
+        description = ''
+          List of color system to be activated.
+          `ccc-action-toggle_input_mode` toggles in this order.
+          The first one is the default used at the first startup.
+          Once activated, it will keep the previous input mode.
 
-    highlighter = {
-      auto_enable = helpers.defaultNullOpts.mkBool false ''
-        Whether to enable automatically on `BufEnter`.
-      '';
+          default:
```

This should updated to match #1530

```suggestion
          Plugin default:
```